pkg/http/http.go

```diff
@@ -0,0 +1,76 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	nhttp "net/http"
+	"time"
+
+	mux "github.com/gorilla/mux"
+)
+
+// Route is an HTTP route
+type Route struct {
+	Path    string
+	Method  string
+	Handler nhttp.Handler
+}
+
+// Server is the app server implementation
+type Server struct {
+	srv    *nhttp.Server
+	router *mux.Router
+	routes *[]Route
+}
+
+// NewServer creates a new server
+func NewServer(port int, routes *[]Route) (*Server, error) {
```

Comment:
This should be replaced by go-lib/http import, right?

Reply:
Or I guess you wanted to avoid this public repo to depend on our private go-lib repo.

Reply:
Correct, coupling should be avoided between the two (so I sacrificed replication of code :) ).